### PR TITLE
Bugfixes for failures uploading data frames via the httpapi found in the ArrayOpR unit tests

### PR DESCRIPTION
**Pull request by jwimberl**
The ArrayOpR unit tests fail against SciDBR 4.0.4 when uploading data frames for two reasons:

* wildcard arguments in the upload method are forwarded to all the dispatched `.UploadDf.httpapi`, `.UploadRaw.httpapi`, etc. methods, but not all captured; they should either not be forwarded when none or needed or else the superfluous arguments should be captured and ignored with an ellipsis
* uploading a data frame to a temporary array with `use_aio_input = TRUE` fails because the temporary array is created with the regular `input` schema and not the `aio_input` schema; this is fixed by deducing the schema (which depends in part on the number of SciDB instances) via lazy evaluation, as is done in the shim path

With the fixes in this branch, the ArrayOpR unit tests all pass against the httpapi.

### CODE REVIEW

**Comment by jwimberl** [comment]
The GitHub Actions workflow must be adjusted in a separate PR to use the newer SciDB 23.10 docker image in place of the deprecated 23.2 image. I ran the tests manually on a SciDB 23.10 machine and there was one failure, but apparently unrelated as it occurred with the shim connection and involved garbage collection:

```
── Failed tests ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
Error (test-api.R:624:3): Test Shim connection
Error in `run_tests_with_gc(shimdb)`: These arrays were created during the test and were not cleaned up: Rarrayop_288ef21bf148a7
Backtrace:
    ▆
 1. └─scidb (local) run_tests_with_gc(shimdb) at test-api.R:624:2

[ FAIL 1 | WARN 0 | SKIP 0 | PASS 252 ]
```

File: R/httpapi.R [hunk 1]
```diff
@@ -747,11 +747,11 @@ Upload.httpapi <- function(db,
   desc@dim_chunk_sizes <- as.numeric(dim_chunk_sizes %||% chunk_size) %||% NULL
 
   if (inherits(payload, "raw")) {
-    .UploadRaw.httpapi(db, payload, desc, ...)
+    .UploadRaw.httpapi(db, payload, desc)
```
Comment (rls-p4) [inline R/httpapi.R:750]: Wouldn't it be more consistent to add `...` arguments to the `.UploadRaw()` and `.UploadSparseMatrix()` functions (and have them ignore it)? That way if we need to add special options for those functions, the calls here won't need to change.

Comment (jwimberl) [reply R/httpapi.R:750]: Sure, I'll make that change

Comment (jwimberl) [reply R/httpapi.R]: Changes pushed

**Comment by jwimberl** [comment]
> I ran the tests manually on a SciDB 23.10 machine and there was one failure, but apparently unrelated as it occurred with the shim connection and involved garbage collection:

Oh, I realize now this is almost certainly related to the environment and not a rare spurious issue in SciDBR; the GC unit test logic assumes that no arrays are being simultaneously created in the database by other connections (true for the Actions workflow but not the local test I ran), so this is irrelevant.

**Review by rls-p4 — APPROVED** [review]
Looks good to me

**Comment by jwimberl** [comment]
> The GitHub Actions workflow must be adjusted in a separate PR to use the newer SciDB 23.10 docker image in place of the deprecated 23.2 image

As it turns out the workflow was future proofed and only a change to a repository variable was required to use the new SciDB 23.10 docker image. All tests now pass.

**Pull request merged**